lib/dry/logic/predicates.rb

```diff
@@ -148,11 +148,19 @@ def self.included(other)
       end
 
       predicate(:includes?) do |value, input|
-        input.include?(value)
+        begin
+          case input
+          when String, Range, Array, Hash then input.include?(value)
```

Comment:
I wonder if we could just check if `respond_to?(:include?)` instead of listing these types?